internal/service/base/resource_language_translation.go

```diff
@@ -0,0 +1,507 @@
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/patrickcping/pingone-go-sdk-v2/management"
+	"github.com/pingidentity/terraform-provider-pingone/internal/framework"
+	"github.com/pingidentity/terraform-provider-pingone/internal/framework/customtypes/pingonetypes"
+	"github.com/pingidentity/terraform-provider-pingone/internal/sdk"
+	"github.com/pingidentity/terraform-provider-pingone/internal/verify"
+)
+
+var (
+	_ resource.Resource                = &languageTranslationResource{}
+	_ resource.ResourceWithConfigure   = &languageTranslationResource{}
+	_ resource.ResourceWithImportState = &languageTranslationResource{}
+)
+
+func NewLanguageTranslationResource() resource.Resource {
+	return &languageTranslationResource{}
+}
+
+type languageTranslationResource serviceClientType
+
+func (r *languageTranslationResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_language_translation"
+}
+
+func (r *languageTranslationResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	resourceConfig, ok := req.ProviderData.(framework.ResourceType)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected the provider client, got: %T. Please report this issue to the provider maintainers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.Client = resourceConfig.Client.API
+	if r.Client == nil {
+		resp.Diagnostics.AddError(
+			"Client not initialised",
+			"Expected the PingOne client, got nil.  Please report this issue to the provider maintainers.",
+		)
+		return
+	}
+}
+
+type languageTranslationResourceModel struct {
+	EnvironmentId  pingonetypes.ResourceIDValue `tfsdk:"environment_id"`
+	Id             pingonetypes.ResourceIDValue `tfsdk:"id"`
+	Key            types.String                 `tfsdk:"key"`
+	Locale         types.String                 `tfsdk:"locale"`
+	ReferenceText  types.String                 `tfsdk:"reference_text"`
+	ShortKey       types.String                 `tfsdk:"short_key"`
+	TranslatedText types.String                 `tfsdk:"translated_text"`
+}
+
+func (r *languageTranslationResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+
+	resp.Schema = schema.Schema{
+		Description: "Resource to create and manage the language translation.",
```

Comment:
"a language translation" is probably better than "the" (I realize this most likely came from my codegen thinking it was a singleton)